WordPress/Classes/Networking/Pinghub.swift

```diff
@@ -218,13 +221,85 @@ internal protocol Socket: AnyObject {
 // MARK: - Starscream
 
 private func starscreamSocket(url: URL, token: String) -> Socket {
-    var request = URLRequest(url: PinghubClient.endpoint)
+    var request = URLRequest(url: url)
     request.setValue("Bearer \(token)", forHTTPHeaderField: "Authorization")
-    return WebSocket(request: request)
+    return StarscreamWebSocket(socket: WebSocket(request: request))
```

Comment:
By default, this `WebSocket` initialiser's `useCustomEngine` parameter is true. We could change it to be`WebSocket(request: request, useCustomEngine: false)` which leads Starscream to use `URLSession.webSocket(with:)`. The app works out exactly the same based on my testing.

However, that change causes the new unit tests to fail. I suspect it's an implementation issue in Starscream's `WebSocketServer`. I didn't look into it further, thus didn't change this initializer.

Reply:
I guess whether to look more into this will depend on whether we decide to remove Starscream.